test/test_controller.js

```diff
@@ -0,0 +1,364 @@
+            function (inDroplist) {
+                assert.isFalse(inDroplist.valueOf(), "Excpect Account 0 not to be in airdrop list");
+                return controller_instance.airdropList.call(addAccounts[1]);
+            }).then(
+            function (inDroplist) {
+                assert.isFalse(inDroplist.valueOf(), "Excpect Account 1 not to be in airdrop list");
+            });
+    });
+});
+
+contract('ICO Controller', function (accounts) {
+    it("test getAirdropTokens", function () {
+        var addAccounts = [accounts[3], accounts[4]];
+        var controller_instance;
+        var token_instance;
+        var airdropTokensSupply;
+        var expectAirdropTokensSupply = 5000000000000000000000000;
+        return Controller.deployed().then(
+            function (inst) {
+                controller_instance = inst;
+                return controller_instance.addAirdrop(addAccounts);
+            }).then(
+            function () {
+                // stat ICOs
+                return controller_instance.startPrivateOffer(
+                    Math.ceil(Date.now() / 1000), Math.ceil(Date.now() / 1000));
+            }).then(
+            function () {
+                wait(2);
+                return controller_instance.startPreSaleIco(
+                    Math.ceil(Date.now() / 1000), Math.ceil(Date.now() / 1000));
+            }).then(
+            function () {
+                wait(2);
+                return controller_instance.startCrowdsale(
+                    Math.ceil(Date.now() / 1000), Math.ceil(Date.now() / 1000));
+            }).then(
+            function () {
+                wait(20);
+                return controller_instance.finishCrowdsale();
+            }).then(
+            function () {
+                return controller_instance.getAirdropTokens();
+            }).then(
+            function () {
+                assert.isFalse(true, "Expect access exception. The function is only for airdrop accounts")
+            }).catch(
+            function (error) {
+                return controller_instance.getAirdropTokens({from: addAccounts[0]});
+            }).then(
+            function () {
+                return controller_instance.token();
+            }).then(
+            function (token_addr) {
+                token_instance = Token.at(token_addr);
+                return controller_instance.AIRDROP_SUPPLY.call();
+            }).then(
+            function (amount) {
+                airdropTokensSupply = amount.valueOf();
+                assert.equal(airdropTokensSupply, expectAirdropTokensSupply, "Wrong amount of airdrop tokens");
+                return token_instance.balanceOf(addAccounts[0]);
+            }).then(
+            function (balance) {
+                assert.equal(balance.valueOf(), expectAirdropTokensSupply / addAccounts.length, "Wrong amount of account token balance");
+            });
+    });
+
+});
+
+contract('ICO Controller', function (accounts) {
```

Comment:
In my opinion, if you split tests into blocks, the names of the blocks should be different.